src/LightSpeed/userFunction.js

```diff
@@ -53,33 +53,100 @@ const COMMON_SCOPE = {
-    // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
-    ticksSinceSpawn: 0,
+// Set method that checks type for number properties of objects
+function propNumFactory(obj, propName, valName) {
+    Object.defineProperty(obj, propName, {
+        set(newVal) {
+            if (Number.isNaN(Number(newVal)))
+                throw new TypeError(`${propName} must be a number.`);
+            else if (Number(newVal) < 0)
+                throw new RangeError(`${propName} must be a positive number.`);
+            valName = newVal; // eslint-disable-line no-param-reassign
+        },
+        get() {
+            return valName;
+        },
+    });
+}
+
+const NUMTYPE_HANDLER = {
+    set(obj, prop, value) {
+        if (prop in obj) {
+            if (Number.isNaN(Number(value)))
+                throw new TypeError(`Value for ${prop} must be a number.`);
+            else
+                return Reflect.set(obj, prop, value);
+        } else {
+            throw new TypeError(`Unknown property ${prop}.`);
+        }
+    },
+
+    get(target, name, receiver) {
+        if (name in target)
+            return Reflect.get(target, name, receiver);
+
+        throw new TypeError(`Unknown property ${name}`);
+    },
 };
 
+// Proxy for position values
+const POSITION = {x: 0, y: 0};
+const POS_PROXY = new Proxy(POSITION, NUMTYPE_HANDLER);
+
 const COMMON_UPDATE_SCOPE = {
     // Keep this in sync with UpdateEnemyScope class in
     // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
     playerShip: {
-        position: {x: 0, y: 0},
+        position: POS_PROXY,
     },
     enemy: {
-        position: {x: 0, y: 0},
-        velocity: {x: 0, y: 0},
+        position: POS_PROXY,
+        velocity: POS_PROXY,
         data: {},
     },
 };
 
+let invulnerableTimer;
+let shrinkTimer;
+let attractTimer;
+const POWERUP = {
+    ship: {
+        position: POS_PROXY,
+        invulnerableTimer: 0,
+        shrinkTimer: 0,
+        attractTimer: 0,
+    },
+    powerUpType: [
+        'blowup',
+        'invulnerable',
+        'upgrade',
+    ],
+    blowUpEnemies() {},  // eslint-disable-line no-empty-function
+};
+
+let ticksSinceSpawn;
+var COMMON_SPAWN_SCOPE = {
+    // Keep this in sync with SpawnScope class in
+    // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
+    ticksSinceSpawn: 0,
+};
+
 const USER_FUNCTIONS = {
     spawnEnemy: {
         name: 'spawnEnemy',
         args: [],
         modelProp: 'spawnEnemyCode',
         perLevel: true,
         getScope() {
-            return Object.assign({}, COMMON_SPAWN_SCOPE, COMMON_SCOPE);
+            const ENEMY_SCOPE = Object.assign({}, COMMON_SPAWN_SCOPE, COMMON_SCOPE);
+            propNumFactory(ENEMY_SCOPE, 'ticksSinceSpawn', ticksSinceSpawn);
+            return ENEMY_SCOPE;
+        },
+        validate(func) {
+            const ret = func();
+            // Don't throw an error when returning null
+            if (ret && !COMMON_SCOPE.enemyTypes.includes(ret))
```

Comment:
Huh, I see what you mean in the other instance of this check where you have to allow `undefined`, since we have an empty `handlePowerup` function in many Lightspeed levels ... we'd either have to change all the defaults to `return null` or something like that. Given that, I think what you have is fine for the time being.